README.md

```diff
@@ -0,0 +1,103 @@
+
+## Instruction
+To use the program you need to download the right executable file depending on which system you work. Then you should open terminal and go to directory with newly
+downloaded file. To run it you should type something like this on windows:  join {path1} {path2} {column_name} {join_type}(optional) 
+
+## Description of main issues
+According to definition of csv format:
+* each record is in a separate line
+* last record may not have ending line break
+* each line consists of the same number of fields separated by commas
+* field may be enclosed with double quotes but if it's not there can't be double quotes inside
+* fields containing commas, line breaks and double quotes must be enclosed with double quotes
+* double quotes inside field enclosed with double quotes must be preceded with another
+
+One challenge was to parse the lines properly. Algorithm solving this problem had to recognize fields enclosed with double quotes and also
+inner commas and other double quotes. Solving this problem was possible by simply counting the number of double quotes in current field
+and the fact that number of ending quote is always even and the next character is separating comma. 
+Code:
+```python
+    def parse_line(line: str) -> list:
+        line = line.rstrip()
+        start_i = 0
+        items = []
+        quotes = 0
+        for i, char in enumerate(line):
+            if char == ',' and not quotes:
+                items.append(line[start_i:i])
+                start_i = i + 1
+
+            if char == '"':
+                quotes += 1
+
+            try:
+                if line[i + 1] == ',' and quotes % 2 == 0:
+                    quotes = 0
+            except IndexError:
+                break
+
+        items.append(line[start_i:])
+        return items
+```
+
+Next issue was to properly join files. My first idea was about using lists to store lines, but code wasn't clear and it was hard to operate on lists
+when it came to settling about the order of printing. Next version included dictionaries to be able to easily acces fields in any order. At the beginning of the files
+headers are loaded to dictionaries as a keys and then every line is parsed and then fields are put into dictionary as a values to matching keys.
+To match the same values in specified column in two files I used simple nested loops and when same values from two files meet the line is printed and that is all for inner join.
+Left and right join are almost the same but different files are the "main" file so I decided to implement right join as left join but switching the paths before reading files.
+To make left join or "my right join" I just added a new bool which is set to True only when there is a match in other file, so when its value is False after finishing the inner
+loop it means that there is no match and missing values are set to Null. Here is a code:
+```python
+            if join_type == 'right':
+                path1, path2 = path2, path1
```

Comment:
You have described the first dataset as main. It would be more descriptive to use the same name here, e.g.
```
if join_type == 'right':
  main, other = path2, path1
else:
  main, other = path1, path2
```

Of course the `other` name could also be improved somehow but nothing comes to mind